src/internal.c

```diff
@@ -29285,13 +29438,21 @@ int SendClientKeyExchange(WOLFSSL* ssl)
                     peerKey = (ssl->specs.static_ecdh) ?
                               ssl->peerEccDsaKey : ssl->peerEccKey;
 
-                    ret = EccSharedSecret(ssl,
-                        (ecc_key*)ssl->hsKey, peerKey,
-                        args->encSecret + OPAQUE8_LEN, &args->encSz,
-                        ssl->arrays->preMasterSecret,
-                        &ssl->arrays->preMasterSz,
-                        WOLFSSL_CLIENT_END
-                    );
+                    #ifdef WOLFSSL_MAXQ10XX_TLS
+                    if ((ssl->hsKey) &&
+                        (((ecc_key*)ssl->hsKey)->maxq_ctx.hw_storage == 1)) {
+                        WOLFSSL_MSG("Skipping EccSharedSecret()");
```

Comment:
Skipping shared secret calc because it is done in. HW? Can we do this in a more generic way like a certain new PK callback is setup?

Reply:
I came up with the idea of setting `EccSharedSecretCb` to a callback that does nothing.  But there are a few instances of : 
```
                    if (ssl->ctx->EccSharedSecretCb != NULL) {
                        break;
                    }
```

Which made it really confusing and hard to debug. 

I should have read your comment more carefully.  You are suggesting a new callback.  I can add a new one called `EccSkipSharedSecret` . It would indicate whether or not to call `EccSharedSecret()`

Reply:
New callback created.